drivers/ethernet/eth_litex_liteeth.c

```diff
@@ -0,0 +1,314 @@
+		goto out;
+	}
+
+	/* copy data to buffer */
+	if (net_pkt_write(pkt, (void *)config->rx_buf[rxslot], len) != 0) {
+		LOG_ERR("Failed to append RX buffer to context buffer");
+		net_pkt_unref(pkt);
+		goto out;
+	}
+
+	/* receive data */
+	r = net_recv_data(context->iface, pkt);
+	if (r < 0) {
+		LOG_ERR("Failed to enqueue frame into RX queue: %d", r);
+		net_pkt_unref(pkt);
+	}
+
+out:
+	irq_unlock(key);
+}
+
+static void eth_irq_handler(const struct device *port)
+{
+	const struct eth_litex_config *config = port->config;
+	/* check sram reader events (tx) */
+	if (litex_read8(config->tx_ev_pending_addr) & BIT(0)) {
+		/* TX event is not enabled nor used by this driver; ack just
+		 * in case if some rogue TX event appeared
+		 */
+		litex_write8(BIT(0), config->tx_ev_pending_addr);
+	}
+
+	/* check sram writer events (rx) */
+	if (litex_read8(config->rx_ev_pending_addr) & BIT(0)) {
+		eth_rx(port);
+
+		/* ack writer irq */
+		litex_write8(BIT(0), config->rx_ev_pending_addr);
+	}
+}
+
+static int eth_set_config(const struct device *dev, enum ethernet_config_type type,
+			  const struct ethernet_config *config)
+{
+	struct eth_litex_dev_data *context = dev->data;
+	int ret = -ENOTSUP;
+
+	switch (type) {
+	case ETHERNET_CONFIG_TYPE_MAC_ADDRESS:
+		memcpy(context->mac_addr, config->mac_address.addr, sizeof(context->mac_addr));
+		ret = net_if_set_link_addr(context->iface, context->mac_addr,
+					   sizeof(context->mac_addr), NET_LINK_ETHERNET);
+		break;
+	default:
+		break;
+	}
+
+	return ret;
+}
+
+static int eth_start(const struct device *dev)
+{
+	const struct eth_litex_config *config = dev->config;
+
+	litex_write8(1, config->rx_ev_enable_addr);
+
+	litex_write8(BIT(0), config->tx_ev_pending_addr);
+	litex_write8(BIT(0), config->rx_ev_pending_addr);
+
+	return 0;
+}
+
+static int eth_stop(const struct device *dev)
+{
+	const struct eth_litex_config *config = dev->config;
+
+	litex_write8(0, config->rx_ev_enable_addr);
+
+	return 0;
+}
+
+static void eth_iface_init(struct net_if *iface)
+{
+	const struct device *port = net_if_get_device(iface);
+	struct eth_litex_dev_data *context = port->data;
+
+	/* set interface */
+	if (context->iface == NULL) {
+		context->iface = iface;
+	}
+
+	/* initialize ethernet L2 */
+	ethernet_init(iface);
+
+	/* set MAC address */
+	if (net_if_set_link_addr(iface, context->mac_addr, sizeof(context->mac_addr),
+			     NET_LINK_ETHERNET) < 0) {
+		LOG_ERR("setting mac failed");
+		return;
+	}
+}
+
+static enum ethernet_hw_caps eth_caps(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return
+#ifdef CONFIG_NET_VLAN
+		ETHERNET_HW_VLAN |
+#endif
+		ETHERNET_LINK_10BASE_T | ETHERNET_LINK_100BASE_T | ETHERNET_LINK_1000BASE_T;
+}
+
+static const struct ethernet_api eth_api = {
+	.iface_api.init = eth_iface_init,
+	.start = eth_start,
+	.stop = eth_stop,
+	.get_capabilities = eth_caps,
+	.set_config = eth_set_config,
+	.send = eth_tx
+};
+
+#define ETH_LITEX_SLOT_SIZE 0x0800
```

Comment:
This is configurable on the RTL side, right? If so - I'd prefer not to have hardcoded properties in the driver and this define should be moved to the devicetree.

Reply:
in litex this is hard-coded to `2**bits_for(eth_mtu)`, with `eth_mcu = 1530` so:
`2**((1530).bit_length())` in python, which is `2048` or `0x800`.
see https://github.com/enjoy-digital/liteeth/blob/583137eaf3a8c8fa5ca6e32ddec92f45b741ab75/liteeth/mac/__init__.py#L62

I don't see that changing anytime soon.

Reply:
kk, makes sense